Python/Product/TestAdapter/PythonStackTraceParser.cs

```diff
@@ -39,17 +55,68 @@ sealed class PythonStackTraceParser : IStackTraceParser {
                     // In some cases (django), we may get a file name that isn't really a file
                     // Ex: File "<frozen importlib._bootstrap>", line 978, in _gcd_import
                     var file = match.Groups[1].Value;
+                    string funcName = match.Groups[3].Value;
                     if (File.Exists(file)) {
                         yield return new StackFrame(
-                            match.Groups[3].Value,
+                            GetQualifiedFunctionName(file, lineno, funcName, languageVersion),
                             file,
                             lineno
                         );
                     } else {
-                        yield return new StackFrame(match.Groups[3].Value);
+                        yield return new StackFrame(funcName);
                     }
                 }
             }
         }
+
+        /// <summary>
```

Comment:
Can you move that comment to `QualifiedFunctionNameWalker`, since it's now defined far away from these helpers? This helper can then just reference that (as well as the one in debugger).